test/test_models.py

```diff
@@ -220,399 +218,389 @@ def get_export_import_copy(m):
+    _check_jit_scriptable(model, (x,), unwrapper=script_model_unwrapper.get(model_name, None))
+
+    if dev == torch.device("cuda"):
+        with torch.cuda.amp.autocast():
+            out = model(x)
+            # See autocast_flaky_numerics comment at top of file.
+            if model_name not in autocast_flaky_numerics:
+                _assert_expected(out.cpu(), model_name, prec=0.1)
+            assert out.shape[-1] == 50
 
 
 @pytest.mark.parametrize('model_name', get_available_segmentation_models())
-@pytest.mark.parametrize('dev', _devs)
+@pytest.mark.parametrize('dev', cpu_and_gpu())
 def test_segmentation_model(model_name, dev):
-    ModelTester()._test_segmentation_model(model_name, dev)
+    set_rng_seed(0)
+    defaults = {
+        'num_classes': 10,
+        'pretrained_backbone': False,
+        'input_shape': (1, 3, 32, 32),
+    }
+    kwargs = {**defaults, **_model_params.get(model_name, {})}
+    input_shape = kwargs.pop('input_shape')
+
+    model = models.segmentation.__dict__[model_name](**kwargs)
+    model.eval().to(device=dev)
+    # RNG always on CPU, to ensure x in cuda tests is bitwise identical to x in cpu tests
+    x = torch.rand(input_shape).to(device=dev)
+    out = model(x)["out"]
+
+    def check_out(out):
+        prec = 0.01
+        try:
+            # We first try to assert the entire output if possible. This is not
+            # only the best way to assert results but also handles the cases
+            # where we need to create a new expected result.
+            _assert_expected(out.cpu(), model_name, prec=prec)
+        except AssertionError:
+            # Unfortunately some segmentation models are flaky with autocast
+            # so instead of validating the probability scores, check that the class
+            # predictions match.
+            expected_file = _get_expected_file(model_name)
+            expected = torch.load(expected_file)
+            torch.testing.assert_close(out.argmax(dim=1), expected.argmax(dim=1), rtol=prec, atol=prec)
+            return False  # Partial validation performed
+
+        return True  # Full validation performed
+
+    full_validation = check_out(out)
+
+    _check_jit_scriptable(model, (x,), unwrapper=script_model_unwrapper.get(model_name, None))
+
+    if dev == torch.device("cuda"):
+        with torch.cuda.amp.autocast():
+            out = model(x)["out"]
+            # See autocast_flaky_numerics comment at top of file.
+            if model_name not in autocast_flaky_numerics:
+                full_validation &= check_out(out)
+
+    if not full_validation:
+        msg = "The output of {} could only be partially validated. " \
+              "This is likely due to unit-test flakiness, but you may " \
+              "want to do additional manual checks if you made " \
+              "significant changes to the codebase.".format(test_segmentation_model.__name__)
+        warnings.warn(msg, RuntimeWarning)
+        pytest.skip(msg)
```

Comment:
Since we no longer use `unittest`, it is now not possible to get the function name through the internal unittest method (`self._testMethodName`) which was prior to the change.

If there is a better way to achieve the same, please share.

Reply:
That sounds just fine!

Reply:
Cool!